### ISSUE TITLE
torch.mv backward has a performance cliff on CPU

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#44739]
## 🐛 Bug

The derivative formula for torch.mv (for vec) is [`self.t().mv(grad)`](https://github.com/pytorch/pytorch/blob/69839ea3f6680d068db38795afe03ae4b8a30962/tools/autograd/derivatives.yaml#L747). However, torch.mv (on CPU) has bad performance for non-contiguous tensors because it will always [copy non-contiguous tensors](https://github.com/pytorch/pytorch/blob/69839ea3f6680d068db38795afe03ae4b8a30962/aten/src/ATen/native/Blas.cpp#L35).

However, the torch.mm CPU implementation does not have this problem. Simply using `self.t().mm(grad.unsqueeze(-1)).squeeze(-1)` is more performant (see next section). The proposal here is to improve the implementation of mv on CPU to better handle non-contiguous inputs and to check this via benchmarking.

## To Reproduce

(in ipython)
```
def time_mv_backward(n, m):
    M = torch.randn(m, n)
    v = torch.randn(m)
    %timeit M.t().mv(v)
    %timeit M.t().mm(v.unsqueeze(-1)).squeeze(-1)

[ins] In [8]: time_mv_backward(100, 100)
20.1 µs ± 139 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
8.88 µs ± 36.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

[ins] In [9]: time_mv_backward(1000, 1000)
2.57 ms ± 80.3 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
40.2 µs ± 2.57 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)

[ins] In [10]: time_mv_backward(10000, 10000)
484 ms ± 2.27 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
6.77 ms ± 1.34 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

## Expected behavior

mv_backward shouldn't be slower than the alternative!

## Environment

PyTorch master

cc @vincentqb @vishwakftw @jianyuh @nikitaved @pearu @VitalyFedyunin @ngimel

**Comment by ngimel** [comment]
This was due to incorrect lda_cond that was forcing extra copies, being fixed in #44681

**Comment by ngimel** [comment]
I can't reproduce on master built with mkl, even with the extra copy the difference can't be that large. 
<details>
<summary>script</summary>

```
import torch
from torch.utils._benchmark import Timer
from torch.utils._benchmark import Compare
import sys

results = []
num_threads = 1
sizes = (100, 1000, 10000)
for size in sizes:
    print(size)
    m = torch.randn(size, size, device="cpu")
    v = torch.randn(size)

    tasks = [("torch.mv(m.t(), v)", "mv"),
               ("(torch.mm(m.t(), v.unsqueeze(-1))).squeeze(-1)", "mm")]
    timers = [Timer(stmt=stmt, num_threads=num_threads, label="mv_backward", sub_label=f"{size}",
    description=label, globals=globals()) for stmt, label in tasks]
    repeats = 3

    for i, timer in enumerate(timers * repeats):
        results.append(
            timer.blocked_autorange()
        )
        print(f"\r{i + 1} / {len(timers) * repeats}", end="")
        sys.stdout.flush()
print()

comparison = Compare(results)
comparison.print()
```
</details>

Results:
```
[---------- mv_backward ----------]
             |     mv    |     mm  
1 threads: ------------------------
      100    |      9.1  |     11.8
      1000   |    131.1  |    135.2
      10000  |  41081.5  |  41048.3

Times are in microseconds (us).
```

**Comment by zou3519** [comment]
Nice, it looks like #44681 will fix it. Maybe we should investigate the performance of contiguous (does it seem slower than it should be?)

**Comment by zou3519** [comment]
Fixed in #44681 by @ngimel!

We can file another issue for contiguous if we think that it is slower than it should be.
